### PR TITLE
refactoring func invokeFunction

### PR DESCRIPTION
**Pull request by luanasoares0901**
### Description

Applying clean code to the invokeFunction function. The function was too large and had complexity problems, so functions and variables were renamed to better describe their purpose. In terms of modularity, the function was split into several smaller functions to improve readability and reusability. Finally, error handling has been assigned to a specific section.

### CODE REVIEW

**Comment by bkimminich** [comment]
Hi! I assume that this PR stems from some university course assignment in Clean Code software development? I'm all for that in theory, because I advocated and taught Clean Code myself for many years. In practice, this PR is a complete train wreck. Let me explain:

The code changes MUST have been submitted via PR without EVER testing locally. I know this because the CI/CD pipelines broke in literally all possible places at once. Tests broke, meaning that your code changes were not functionally neutral. You introduced 83 code linter errors, because your refactoring did not care about our code style conventions at all. You broke our Refactoring Safely Net (RSN) for one coding challenge, because you very likely weren't aware that this is something we even have.

In summary, this is a great example how even the most well meant actions can lead to the most ill effects. Clean Code requires knowledge of the context of the software. It also requires to follow the project team's guidelines, which is even mentioned in a chapter in Uncle Bob's book.

Most of my complaints could have been prevented _by reading the contribution guidelines_ of the project, as those explain how to fully test locally, run the linters, run the RSN and more:

https://pwning.owasp-juice.shop/companion-guide/latest/part3/contribution.html

Long story short, this PR is unmergable as it is. If you'd like to give it a second try following the above mentioned guidelines, I'm happy to manually review it and give further feedback.
